File: implementations/typescript/packages/environment/test/environment.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { ENVIRONMENT_EVENT_TYPES, isEnvironmentEventType } from "../src/index.js";

test("environment event types includes all 4 registry entries", () => {
  const expected = [
    "environment.observed",
    "environment.changed",
    "environment.alerted",
    "environment.error"
  ];
  for (const type of expected) {
    assert.equal(ENVIRONMENT_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(ENVIRONMENT_EVENT_TYPES.size, expected.length);
});

test("isEnvironmentEventType positives", () => {
  assert.equal(isEnvironmentEventType("environment.observed"), true);
  assert.equal(isEnvironmentEventType("environment.changed"), true);
  assert.equal(isEnvironmentEventType("environment.alerted"), true);
  assert.equal(isEnvironmentEventType("environment.error"), true);
});

test("isEnvironmentEventType negatives", () => {
  assert.equal(isEnvironmentEventType("task.submitted"), false);
  assert.equal(isEnvironmentEventType("session.opened"), false);
  assert.equal(isEnvironmentEventType("context.updated"), false);
  assert.equal(isEnvironmentEventType(""), false);
});

File: implementations/go/environment/environment_test.go
package environment

import "testing"

func TestEnvironmentEventTypesIncludesAll4RegistryEntries(t *testing.T) {
	expected := []string{
		"environment.observed",
		"environment.changed",
		"environment.alerted",
		"environment.error",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("expected %d types, got %d", len(expected), len(EventTypes))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	for _, typ := range []string{
		"environment.observed",
		"environment.changed",
		"environment.alerted",
		"environment.error",
	} {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	for _, typ := range []string{
		"task.submitted",
		"session.opened",
		"context.updated",
		"",
	} {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}
